integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -353,25 +348,42 @@ def _stream_response_and_convert_to_messages(
         """
         text = ""
         tool_calls = []
-        last_metadata = None
+        chunk_dict = {}
 
         for chunk in stream:
+            content_to_stream = ""
             chunk_dict = chunk.to_dict()
-            last_metadata = chunk_dict
+
             # Only one candidate is supported for chat functionality
             candidate = chunk_dict["candidates"][0]
 
             for part in candidate["content"]["parts"]:
-                if part.get("text"):
-                    text += part["text"]
-                elif part.get("function_call"):
+                if new_text := part.get("text"):
+                    content_to_stream += new_text
+                    text += new_text
+                elif new_function_call := part.get("function_call"):
+                    content_to_stream += json.dumps(dict(new_function_call))
                     tool_calls.append(
                         ToolCall(
-                            tool_name=part["function_call"]["name"],
-                            arguments=dict(part["function_call"]["args"]),
+                            tool_name=new_function_call["name"],
+                            arguments=dict(new_function_call["args"]),
```

Comment:
do we need these two dict wrappers?

Reply:
thanks for asking...

I've verified that in the `_stream_response_and_convert_to_messages` method we don't need `dict` because we are already using a dictionary (`chunk.to_dict()`).

It was probably a copy-paste from `_convert_response_to_messages`, where we need `dict`, otherwise `part.function_call.args` is a `proto.marshal.collections.maps.MapComposite` object.